HopsanGUI/Dialogs/OptimizationScriptWizard.cpp

```diff
@@ -0,0 +1,1611 @@
+    pSettingsLayout->addWidget(mpBetaLabel,            row,   0);
+    pSettingsLayout->addWidget(mpBetaLineEdit,         row++, 1);
+    pSettingsLayout->addWidget(mpC1Label,              row,   0);
+    pSettingsLayout->addWidget(mpC1LineEdit,           row++, 1);
+    pSettingsLayout->addWidget(mpGammaLabel,           row,   0);
+    pSettingsLayout->addWidget(mpGammaLineEdit,        row++, 1);
+    pSettingsLayout->addWidget(mpRhoLabel,             row,   0);
+    pSettingsLayout->addWidget(mpRhoLineEdit,          row++, 1);
+    pSettingsLayout->addWidget(mpSigmaLabel,           row,   0);
+    pSettingsLayout->addWidget(mpSigmaLineEdit,        row++, 1);
+    pSettingsLayout->addWidget(mpC2Label,              row,   0);
+    pSettingsLayout->addWidget(mpC2LineEdit,           row++, 1);
+    pSettingsLayout->addWidget(mpVmaxLabel,            row,   0);
+    pSettingsLayout->addWidget(mpVmaxLineEdit,         row++, 1);
+    pSettingsLayout->addWidget(mpFLabel,               row,   0);
+    pSettingsLayout->addWidget(mpFLineEdit,            row++, 1);
+    pSettingsLayout->addWidget(mpCRLabel,              row,   0);
+    pSettingsLayout->addWidget(mpCRLineEdit,           row++, 1);
+    pSettingsLayout->addWidget(mpCPLabel,              row,   0);
+    pSettingsLayout->addWidget(mpCPLineEdit,           row++, 1);
+    pSettingsLayout->addWidget(mpMPLabel,              row,   0);
+    pSettingsLayout->addWidget(mpMPLineEdit,           row++, 1);
+    pSettingsLayout->addWidget(mpNumModelsLabel,       row,   0);
+    pSettingsLayout->addWidget(mpNumModelsLineEdit,    row++, 1);
+    pSettingsLayout->addWidget(mpMethodLabel,          row,   0);
+    pSettingsLayout->addWidget(mpMethodComboBox,       row++, 1);
+    pSettingsLayout->addWidget(mpLengthLabel,          row,   0);
+    pSettingsLayout->addWidget(mpLengthSpinBox,        row++, 1);
+    pSettingsLayout->addWidget(mpPercDiffLabel,        row,   0);
+    pSettingsLayout->addWidget(mpPercDiffLineEdit,     row++, 1);
+    pSettingsLayout->addWidget(mpCountMaxLabel,        row,   0);
+    pSettingsLayout->addWidget(mpCountMaxSpinBox,      row++, 1);
+    pSettingsLayout->addWidget(pEpsilonXLabel,         row,   0);
+    pSettingsLayout->addWidget(mpEpsilonXLineEdit,     row++, 1);
+    pSettingsLayout->addWidget(mpPlotBestWorstCheckBox,row++, 0, 1, 2);
+    pSettingsLayout->addWidget(mpPlotParticlesCheckBox,row++, 0, 1, 2);
+    pSettingsLayout->addWidget(mpPlotEntropyCheckBox,  row++, 0, 1, 2);
+    pSettingsLayout->addWidget(mpExport2CSVBox,        row++, 0, 1, 2);
+    pSettingsLayout->addWidget(mpFinalEvalCheckBox,    row++, 0, 1, 2);
+    pSettingsLayout->addWidget(new QWidget(this),      row++, 0, 1, 2);    //Dummy widget for stretching the layout
+    pSettingsLayout->setRowStretch(row++, 1);
+    QWizardPage *pSettingsWidget = new QWizardPage(this);
+    pSettingsWidget->setLayout(pSettingsLayout);
+    pSettingsWidget->setPalette(gpConfig->getPalette());
+    setAlgorithm(0);
```

Comment:
What algorithm is 0, 
ok saw that its undefined, maybe should have that in the Ops enum.
'NoAlgorithm' or similar

Reply:
It does not mean undefined (there is an "++i" in setAlgorithm). 0 simple means use the first algorithm in the drop-down menu (which is currently Nelder-Mead).

Reply:
ok, I did not mean undefined as undefined variable, I saw that i++, I just mean that the magic value 0 is unclear.
I would be nice if there would be a enum value or named value that represent, "nothing chosen" or similar. 
But not such a big deal